### ISSUE TITLE
Not final check not reported if @SuppressWarnings("HideUtilityClassConstructor") define

### ISSUE THREAD

**Issue by xenoterracide** [checkstyle/checkstyle#15652]
with rules

```xml
    <module name="FinalClass" />
    <module name="HideUtilityClassConstructor" />
```

and only `HideUtilityClassConstructor` supplied, there is no error about `FinalClass` if I were to provide the constructor (which would break the code) then the `FinalClass` error showed up. 

I apologize this may be a configuration issue on my part, I'm not certain.

```java
// © Copyright 2024 Caleb Cushing
// SPDX-License-Identifier: AGPL-3.0-or-later

package com.xenoterracide.jpa.transaction;

import com.xenoterracide.jpa.annotation.TransactionScope;
import com.xenoterracide.jpa.util.Constants;
import java.time.Instant;
import java.time.OffsetDateTime;
import java.time.ZoneId;
import java.time.ZonedDateTime;
import org.jspecify.annotations.NonNull;
import org.springframework.beans.factory.config.BeanFactoryPostProcessor;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.transaction.support.SimpleTransactionScope;

/**
 * The type Transaction bean post processor.
 */
@Configuration
@SuppressWarnings("HideUtilityClassConstructor")
class TransactionScopeConfiguration {

  private static final ZoneId UTC = ZoneId.of("UTC");

  @Bean
  static BeanFactoryPostProcessor transactionBeanPostProcessor() {
    return beanFactory -> beanFactory.registerScope(Constants.TRANSACTION_SCOPE, new SimpleTransactionScope());
  }

  @Bean
  @TransactionScope
  static Instant instantNow() {
    return Instant.now();
  }

  @Bean
  @TransactionScope
  static ZonedDateTime zonedDateTimeNow(@NonNull Instant instantNow) {
    return instantNow.atZone(UTC);
  }

  /**
   * Offset date time now offset date time.
   *
   * @param zonedDateTimeNow the zoned date time now
   * @return the offset date time
   */
  @Bean
  @TransactionScope
  static OffsetDateTime offsetDateTimeNow(@NonNull ZonedDateTime zonedDateTimeNow) {
    return zonedDateTimeNow.toOffsetDateTime();
  }
}
```

```xml
<?xml version="1.0" ?>
<!DOCTYPE module PUBLIC "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
  "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="SuppressWarningsFilter" />

  <!--- PRETTIER START -->

  <module name="BeforeExecutionExclusionFileFilter">
    <property name="fileNamePattern" value="module\-info\.java$" />
  </module>
  <module name="TreeWalker">
    <module name="OuterTypeFilename" />
    <module name="NoLineWrap">
      <property name="tokens" value="PACKAGE_DEF, IMPORT, STATIC_IMPORT" />
    </module>
    <module name="LeftCurly" />
    <module name="RightCurly" />
    <module name="WhitespaceAfter" />
    <module name="WhitespaceAround">
      <message
        key="ws.notFollowed"
        value="WhitespaceAround: ''{0}'' is not followed by whitespace. Empty blocks may only be represented as '{}' when not part of a multi-block statement (4.1.3)"
      />
      <message key="ws.notPreceded" value="WhitespaceAround: ''{0}'' is not preceded with whitespace." />
      <property name="allowEmptyConstructors" value="true" />
      <property name="allowEmptyLambdas" value="true" />
      <property name="allowEmptyMethods" value="true" />
      <property name="allowEmptyTypes" value="true" />
      <property name="allowEmptyLoops" value="true" />
      <property name="allowEmptyCatches" value="true" />
      <property name="ignoreEnhancedForColon" value="false" />
      <property
        name="tokens"
        value="ASSIGN, BAND, BAND_ASSIGN, BOR, BOR_ASSIGN, BSR, BSR_ASSIGN, BXOR,
                    BXOR_ASSIGN, COLON, DIV, DIV_ASSIGN, DO_WHILE, EQUAL, GE, GT, LAMBDA, LAND,
                    LCURLY, LE, LITERAL_CATCH, LITERAL_DO, LITERAL_ELSE, LITERAL_FINALLY,
                    LITERAL_FOR, LITERAL_IF, LITERAL_RETURN, LITERAL_SWITCH, LITERAL_SYNCHRONIZED,
                    LITERAL_TRY, LITERAL_WHILE, LOR, LT, MINUS, MINUS_ASSIGN, MOD, MOD_ASSIGN,
                    NOT_EQUAL, PLUS, PLUS_ASSIGN, QUESTION, RCURLY, SL, SLIST, SL_ASSIGN, SR,
                    SR_ASSIGN, STAR, STAR_ASSIGN, LITERAL_ASSERT, TYPE_EXTENSION_AND"
      />
    </module>
    <module name="GenericWhitespace">
      <message key="ws.followed" value="GenericWhitespace ''{0}'' is followed by whitespace." />
      <message key="ws.preceded" value="GenericWhitespace ''{0}'' is preceded with whitespace." />
      <message key="ws.illegalFollow" value="GenericWhitespace ''{0}'' should followed by whitespace." />
      <message key="ws.notPreceded" value="GenericWhitespace ''{0}'' is not preceded with whitespace." />
    </module>
    <module name="OneStatementPerLine" />
    <module name="ModifierOrder" />
    <module name="EmptyLineSeparator">
      <property
        name="tokens"
        value="PACKAGE_DEF, IMPORT, STATIC_IMPORT, CLASS_DEF, INTERFACE_DEF, ENUM_DEF,
                    STATIC_INIT, INSTANCE_INIT, CTOR_DEF, VARIABLE_DEF, RECORD_DEF,
                    COMPACT_CTOR_DEF"
      />
      <property name="allowNoEmptyLineBetweenFields" value="true" />
    </module>
    <module name="SeparatorWrap">
      <property name="id" value="SeparatorWrapDot" />
      <property name="tokens" value="DOT" />
      <property name="option" value="nl" />
    </module>
    <module name="SeparatorWrap">
      <property name="id" value="SeparatorWrapComma" />
      <property name="tokens" value="COMMA" />
      <property name="option" value="EOL" />
    </module>

    <module name="CustomImportOrder">
      <property name="sortImportsInGroupAlphabetically" value="true" />
      <property name="separateLineBetweenGroups" value="true" />
      <property name="customImportOrderRules" value="STATIC###THIRD_PARTY_PACKAGE" />
      <property name="tokens" value="IMPORT, STATIC_IMPORT, PACKAGE_DEF" />
    </module>
    <module name="MethodParamPad">
      <property
        name="tokens"
        value="CTOR_DEF, LITERAL_NEW, METHOD_CALL, METHOD_DEF,
                    SUPER_CTOR_CALL, ENUM_CONSTANT_DEF, RECORD_DEF"
      />
    </module>
    <module name="NoWhitespaceBefore">
      <property
        name="tokens"
        value="COMMA, SEMI, POST_INC, POST_DEC, DOT,
                    LABELED_STAT, METHOD_REF"
      />
      <property name="allowLineBreaks" value="true" />
    </module>
    <module name="ParenPad" />
    <module name="AnnotationLocation">
      <property name="allowSamelineMultipleAnnotations" value="false" />
      <property name="allowSamelineSingleParameterlessAnnotation" value="false" />
      <property name="allowSamelineParameterizedAnnotation" value="false" />
    </module>
  </module>

  <!--- PRETTIER END -->

  <module name="TreeWalker">
    <!-- https://checkstyle.sourceforge.io/filters/ -->
    <!-- in case you're dumb again it's CHECKSTYLE.OFF: checkname|checkname -->
    <module name="SuppressionCommentFilter">
      <property name="offCommentFormat" value="CHECKSTYLE.OFF\: ([\w\|]+)" />
      <property name="onCommentFormat" value="CHECKSTYLE.ON\: ([\w\|]+)" />
      <property name="checkFormat" value="$1" />
    </module>

    <!-- https://checkstyle.org/checks/annotation/ checked all 2024 -->
    <module name="MissingDeprecated" />
    <module name="MissingOverride" />

    <!-- https://checkstyle.sourceforge.io/checks/javadoc/ checked all 2024 -->
    <module name="AtclauseOrder" />
    <module name="InvalidJavadocPosition" />
    <module name="JavadocBlockTagLocation" />
    <module name="JavadocContentLocation" />
    <module name="JavadocMethod" />
    <module name="JavadocMissingLeadingAsterisk" />
    <module name="JavadocMissingWhitespaceAfterAsterisk" />
    <module name="JavadocParagraph" />
    <module name="JavadocStyle" />
    <module name="JavadocTagContinuationIndentation">
      <property name="violateExecutionOnNonTightHtml" value="true" />
    </module>
    <module name="JavadocType">
      <property name="allowUnknownTags" value="true" />
    </module>
    <module name="JavadocVariable">
      <property name="scope" value="protected" />
    </module>
    <module name="MissingJavadocMethod" />
    <module name="MissingJavadocPackage" />
    <module name="MissingJavadocType" />
    <module name="NonEmptyAtclauseDescription" />
    <module name="RequireEmptyLineBeforeBlockTagGroup" />
    <module name="SingleLineJavadoc" />
    <module name="SummaryJavadoc" />

    <!-- Checks for Naming Conventions https://checkstyle.org/checks/naming/ -->
    <module name="AbbreviationAsWordInName" />
    <module name="ConstantName" />
    <module name="LocalFinalVariableName" />
    <module name="LocalVariableName" />
    <module name="MemberName" />
    <module name="MethodName" />
    <module name="PackageName" />
    <module name="ParameterName" />
    <module name="StaticVariableName" />
    <module name="TypeName" />
    <module name="ParameterAssignment" />
    <module name="ReturnCount" />
    <module name="StringLiteralEquality" />
    <module name="CatchParameterName" />
    <module name="ClassTypeParameterName">
      <property name="format" value="^[A-Z]+$" />
    </module>
    <module name="InterfaceTypeParameterName">
      <property name="format" value="^[A-Z]+$" />
    </module>
    <module name="MethodTypeParameterName">
      <property name="format" value="^[A-Z]+$" />
    </module>
    <module name="LambdaParameterName" />

    <!-- imports https://checkstyle.org/checks/imports -->
    <module name="AvoidStarImport" />
    <module name="IllegalImport" />
    <!-- defaults to sun.* packages -->
    <module name="RedundantImport" />
    <module name="UnusedImports">
      <property name="processJavadoc" value="false" />
    </module>

    <!-- Metrics https://checkstyle.sourceforge.io/config_metrics.html -->
    <module name="BooleanExpressionComplexity" />
    <module name="ClassDataAbstractionCoupling" />
    <module name="ClassFanOutComplexity" />
    <module name="CyclomaticComplexity" />
    <module name="JavaNCSS" />
    <module name="NPathComplexity" />

    <!-- Checks for Size Violations https://checkstyle.org/checks/sizes/ -->
    <module name="AnonInnerLength" />
    <module name="ExecutableStatementCount">
      <property name="max" value="20" />
    </module>
    <module name="LambdaBodyLength" />
    <module name="MethodCount">
      <property name="maxTotal" value="20" />
    </module>
    <module name="MethodLength">
      <property name="max" value="25" />
    </module>
    <module name="OuterTypeNumber" />
    <module name="ParameterNumber" />
    <module name="RecordComponentNumber" />

    <!-- whitespace https://checkstyle.org/checks/whitespace/ -->
    <module name="OperatorWrap">
      <property name="tokens" value="QUESTION, COLON" />
    </module>
    <module name="OperatorWrap">
      <property
        name="tokens"
        value="EQUAL, NOT_EQUAL, DIV, PLUS, MINUS, STAR, MOD, SR, BSR, GE, GT, SL, LE, LT, BXOR,
     BOR, LOR, BAND, LAND, TYPE_EXTENSION_AND, LITERAL_INSTANCEOF"
      />
      <property name="option" value="eol" />
    </module>
    <module name="EmptyForIteratorPad" />
    <module name="NoWhitespaceAfter">
      <property
        name="tokens"
        value="AT, INC, DEC, UNARY_MINUS, UNARY_PLUS, BNOT, LNOT, DOT, ARRAY_DECLARATOR, INDEX_OP"
      />
    </module>
    <module name="CommentsIndentation" />
    <module name="AvoidEscapedUnicodeCharacters">
      <property name="allowNonPrintableEscapes" value="true" />
      <property name="allowByTailComment" value="true" />
      <property name="allowEscapesForControlCharacters" value="true" />
    </module>
    <module name="AvoidDoubleBraceInitialization" />
    <module name="AvoidNoArgumentSuperConstructorCall" />
    <module name="EmptyCatchBlock" />
    <module name="DeclarationOrder" />
    <module name="DefaultComesLast" />

    <module name="SingleSpaceSeparator" />
    <module name="TypecastParenPad" />
    <module name="MutableException" />

    <module name="UnnecessaryParentheses" />
    <module name="UnnecessarySemicolonAfterOuterTypeDeclaration" />
    <module name="UnnecessarySemicolonAfterTypeMemberDeclaration" />
    <module name="UnnecessarySemicolonInEnumeration" />
    <module name="UnnecessarySemicolonInTryWithResources" />
    <module name="VariableDeclarationUsageDistance" />

    <!-- Modifier Checks                                    -->
    <!-- See https://checkstyle.org/config_modifiers.html -->
    <module name="RedundantModifier" />
    <module name="AbstractClassName" />

    <!-- https://checkstyle.sourceforge.io/checks/blocks/ -->
    <module name="AvoidNestedBlocks">
      <property name="allowInSwitchCase" value="true" />
    </module>
    <module name="EmptyBlock" />
    <module name="NeedBraces">
      <property name="allowSingleLineStatement" value="true" />
    </module>

    <!-- https://checkstyle.org/checks/coding/ -->
    <module name="EqualsAvoidNull" />
    <module name="FallThrough" />
    <module name="EmptyStatement" />
    <module name="EqualsHashCode" />
    <module name="RequireThis">
      <property name="validateOnlyOverlapping" value="false" />
    </module>
    <module name="ExplicitInitialization" />
    <module name="IllegalInstantiation" />
    <module name="IllegalCatch" />
    <module name="IllegalToken" />
    <module name="IllegalType" />
    <module name="InnerAssignment" />
    <module name="MissingSwitchDefault" />
    <module name="MultipleVariableDeclarations" />
    <module name="SimplifyBooleanExpression" />
    <module name="SimplifyBooleanReturn" />
    <module name="ModifiedControlVariable" />
    <module name="MultipleStringLiterals" />
    <module name="NestedForDepth" />
    <module name="NestedIfDepth" />
    <module name="NestedTryDepth" />
    <module name="NoArrayTrailingComma" />
    <module name="NoClone" />
    <module name="NoFinalizer" />
    <module name="PackageDeclaration" />

    <!-- design https://checkstyle.org/checks/design/ -->
    <module name="FinalClass" />
    <module name="HideUtilityClassConstructor" />
    <module name="InnerTypeLast" />
    <module name="InterfaceIsType" />
    <module name="MutableException" />
    <module name="OneTopLevelClass" />
    <module name="ThrowsCount">
      <property name="ignorePrivateMethods" value="false" />
    </module>
    <module name="VisibilityModifier" />

    <!-- Miscellaneous https://checkstyle.org/checks/misc/ -->
    <module name="ArrayTypeStyle" />
    <module name="TodoComment" />
    <module name="UpperEll" />
    <module name="UncommentedMain">
      <property name="excludedClasses" value="^Application|.*Server$" />
    </module>

    <module name="SuppressWarningsHolder" />
  </module>
  <module name="NewlineAtEndOfFile" />

  <module name="Translation" />

  <module name="RegexpSingleline">
    <!-- trailing whitespace -->
    <property name="format" value="\s+$" />
    <property name="minimum" value="0" />
    <property name="maximum" value="0" />
    <property name="message" value="Line has trailing spaces." />
  </module>

  <!-- https://checkstyle.sourceforge.io/checks/javadoc/ -->
  <module name="JavadocPackage" />

  <!-- Miscellaneous https://checkstyle.org/checks/misc/ -->
  <module name="FileLength" />
  <!-- use method count for SRP instead -->
  <module name="LineLength">
    <property name="max" value="120" />
    <property name="fileExtensions" value="java" />
    <property name="ignorePattern" value="^\s+// http" />
  </module>
  <module name="FileTabCharacter" />

  <!-- Checks for Size Violations https://checkstyle.org/checks/sizes/ -->
  <property name="charset" value="UTF-8" />
  <property name="severity" value="error" />

  <!-- whitespace  https://checkstyle.org/checks/whitespace/ -->
  <property name="fileExtensions" value="java, properties, xml" />
</module>
```

```
com.puppycrawl.tools:checkstyle:9.3=checkstyle
```

```

------------------------------------------------------------
Gradle 8.10
------------------------------------------------------------

Build time:    2024-08-14 11:07:45 UTC
Revision:      fef2edbed8af1022cefaf44d4c0514c5f89d7b78

Kotlin:        1.9.24
Groovy:        3.0.22
Ant:           Apache Ant(TM) version 1.10.14 compiled on August 16 2023
Launcher JVM:  21.0.4 (Eclipse Adoptium 21.0.4+7-LTS)
Daemon JVM:    /home/xeno/.asdf/installs/java/temurin-21.0.4+7.0.LTS (no JDK specified, using current Java home)
OS:            Linux 6.6.44-1-MANJARO amd64

```

**Comment by romani** [comment]
Please follow issue template and please use latest version of checkstyle we support only latest version.

**Comment by BenjaminGaelSerrano** [comment]
Hi! I'm a student working on an open-source contribution project. I've analyzed this issue and I'd like to work on a fix for it. Could you please assign it to me?

**Comment by romani** [comment]
No assignments, just start working on it.

Please [read](https://github.com/checkstyle/checkstyle/blob/master/.github/CONTRIBUTING.md#getting-started) and watch videos at [Starting_Development](https://checkstyle.org/beginning_development.html#Starting_Development).

**Comment by romani** [comment]
Issue is not approved , please help to report issue properly to let me approve it

**Comment by BenjaminGaelSerrano** [comment]
I have submitted the PR, but I haven´t received any feedback yet. 
https://github.com/checkstyle/checkstyle/pull/20258

**Comment by romani** [comment]
all checks works independently from each other.

https://checkstyle.org/checks/design/finalclass.html will raise violation only if **private** ctor is created.

as there is not time to report issue clearly, I am closing it, to let somebody see no open issues and create new
